packages/flutter/lib/src/physics/spring_simulation.dart

```diff
@@ -219,23 +225,21 @@ class _OverdampedSolution implements _SpringSolution {
   }
 
   _OverdampedSolution.withArgs(double r1, double r2, double c1, double c2)
-    : _r1 = r1,
-      _r2 = r2,
-      _c1 = c1,
-      _c2 = c2;
+      : _r1 = r1,
+        _r2 = r2,
+        _c1 = c1,
+        _c2 = c2;
 
   final double _r1, _r2, _c1, _c2;
 
   @override
   double x(double time) {
-    return _c1 * math.pow(math.e, _r1 * time) +
-           _c2 * math.pow(math.e, _r2 * time);
+    return _c1 * math.pow(math.e, _r1 * time) + _c2 * math.pow(math.e, _r2 * time);
```

Comment:
this one is one and the one below i think we should keep in the old style (might be able to force the newline by putting `//` at the end of the first line? though then it might align wrong on the next line)

Reply:
Adding the `//` will force the new line, but the alignment an the next line would be off, as you predicted:

```dart
    return _c1 * math.pow(math.e, _r1 * time) + //
        _c2 * math.pow(math.e, _r2 * time);
```

Alternatively, one could rewrite this with helper variables as follows to keep the parallelism:

```dart
    final double summand1 = _c1 * math.pow(math.e, _r1 * time);
    final double summand2 = _c2 * math.pow(math.e, _r2 * time);
    return summand1 + summand2;
```

Depending on how complicated the formula is, this could actually make it more readable...

Would you find any of those formatter-compatible styles acceptable?